### PR TITLE
Fix data_dir polling when there are subdirectories

### PR DESCRIPTION
**Pull request by twaugh**
When there are changes only in subdirectories of `data_dir`, these are never detected.

This change watches the mtimes of all subdirectories as well as the `data_dir`.

### CODE REVIEW

**Comment by Jmainguy** [comment]
Applied this change at $work and can confirm it now shows V1 and V2 images

**Comment by twaugh** [comment]
@mhrivnak This is the PR I mentioned when we spoke last.

**Comment by rbarlow** [comment]
@twaugh oh man, I hadn't noticed this PR and this bug has been driving me crazy! Haha, if only I had seen it sooner.

File: crane/data.py [hunk 1]
```diff
@@ -79,18 +79,41 @@ def monitor_data_dir(app, last_modified=0):
...
+            time.sleep(polling_interval)
+            if not dirs:
+                # The data directory didn't exist before so re-check
+                break
+
+            # Check if the modified time on any directory is newer
+            # than the most recent change we saw
+            try:
+                logger.debug('Checking for new metadata files')
+                most_recent = max(os.stat(dir).st_mtime for dir in dirs)
```
Comment (rbarlow) [inline crane/data.py]: A DRY improvement might be to make this max() thing a function, even though it's a one liner.

**Pull request merged**